### PR TITLE
got: 0.83 -> 0.86

### PR DESCRIPTION
**Pull request by afh**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/got/default.nix [hunk 1]
```diff
@@ -17,12 +17,18 @@ stdenv.mkDerivation rec {
...
 
   preConfigure = lib.optionals stdenv.isDarwin ''
-    # The configure script assumes dependencies on Darwin are install via
+    # The configure script assumes dependencies on Darwin are installed via
     # Homebrew or MacPorts and hardcodes assumptions about the paths of
     # dependencies which fails the nixpkgs configurePhase.
     substituteInPlace configure --replace 'xdarwin' 'xhomebrew'
   '';
 
+  patches = [ ./darwin.patch ];
```
Comment (afh) [inline pkgs/applications/version-management/got/default.nix:26]: Without this patch the buildPhase (linking) fails with missing symbol bsd_getopt

File: pkgs/applications/version-management/got/default.nix [hunk 2]
```diff
@@ -17,12 +17,18 @@ stdenv.mkDerivation rec {
...
+
+  preBuild = ''
+    buildFlagsArray+=(CFLAGS="-DHAVE_STRLCPY=1 -DHAVE_STRLCAT=1 -DHAVE_STRNSTR=1 -DHAVE_STRMODE=1")
+  '';
```
Comment (afh) [inline pkgs/applications/version-management/got/default.nix:30]: Without these buildFlags the buildPhase fails on macOS with conflicting types for strmode.

**Comment by wahjava** [comment]
Result of `nixpkgs-review pr 220891` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>got</li>
  </ul>
</details>

**Review by wahjava — CHANGES_REQUESTED** [review]
It doesn't quite build on GNU/Linux as is, without https://github.com/wahjava/nixpkgs/commit/54104f1a044ef0419158daac6a24296c183ce2e9. Could you please `git-am` it in your branch, and push it ?

**Comment by afh** [comment]
I've applied the patch as you requested, @wahjava.

What are your thoughts on gating the darwin patch in [libbsd](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/libbsd/default.nix#L25) with `lib.optional stdenv.isDarwin`?

**Comment by wahjava** [comment]
Result of `nixpkgs-review pr 220891` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>got</li>
  </ul>
</details>

**Comment by wahjava** [comment]
> I've applied the patch as you requested, @wahjava.
> 
> What are your thoughts on gating the darwin patch in [libbsd](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/libbsd/default.nix#L25) with `lib.optional stdenv.isDarwin`?

I don't have any strong opinions on that. AFAICT, if the functions in the question are already indicated as defined, then don't re-define them, which is fine.

In your commit you were marking those functions as being available (for GNU/Linux) when they weren't.

File: pkgs/applications/version-management/got/default.nix (hunk 1 above)
Comment (wahjava) [reply pkgs/applications/version-management/got/default.nix:28]: Probably should use an `if` here, instead of lib.optional. It's now broken on Darwin :/

**Review by wahjava — CHANGES_REQUESTED** [review]

**Comment by afh** [comment]
Thanks, @wahjava, I'll look into it.

**Review by wahjava — APPROVED** [review]
Looks good to me. Thank you!

**Comment by afh** [comment]
Thanks for the review, @wahjava, much appreciated 🙏  Looking forward to having this merged.

**Comment by wahjava** [comment]
Many thanks for working on it :bow:

**Comment by wegank** [comment]
Superseded by #222170.

**Pull request closed without merging**